### ISSUE TITLE
e2e-parity gate installs HF reference unpinned → non-reproducible verdicts on borderline prompts

### ISSUE THREAD

**Issue by ohdearquant** [ohdearquant/lattice#236]
## Problem

`e2e-parity.yml` installs the HF reference framework **unpinned**:

```yaml
run: pip install torch transformers tokenizers huggingface_hub
```

The reference (HF) greedy output is therefore **not reproducible across runs** — a fresh GitHub runner can pick up a newer `transformers`/`torch` between two runs hours apart. For prompts whose first-token argmax is borderline (notably the 816-token `LONG_PROMPT`, which the script's own comment flags: *"GDN recurrent state accumulation amplifies tiny f32 rounding differences ... greedy output naturally diverges"*), this flips the PASS/FAIL verdict for reasons **unrelated to the lattice change under test**.

## Evidence

Observed on PR #235 (chunked GDN prefill): the gate passed at commit `085ac25f` and failed at `5fa81e42`, with an **identical prompt set** (the long prompt was added in #228, present at both commits) and **identical lattice output**. Local verification on qwen3.5-0.8b proved lattice's first-5 tokens on the long prompt are byte-invariant across chunked-vs-serial, floor-vs-no-floor (the two commits' kernels), and run-to-run. The only moving variable was the HF reference side.

## Proposed fix

1. **Pin the reference versions** in `e2e-parity.yml` (e.g. `pip install "torch==X" "transformers==Y" "tokenizers==Z"`), so the gate is reproducible and a real regression is distinguishable from reference drift.
2. Consider committing the HF reference token IDs as goldens (like the embed parity goldens) so the gate does not need to run HF at all on every PR — only when intentionally refreshing the reference.

## Separate question (not this issue)

On the 816-token prompt, lattice's first generated token differs from HF's (`...main(input_data=None):` vs `...merge_sort(arr):`). Both are plausible continuations and the argmax is borderline, but whether lattice's long-prefill final-position logits have a real accuracy gap vs HF at long context is worth its own investigation. That divergence is present on main's serial path and is independent of any chunked-prefill PR.
